docs/architecture/adr-002-docs-structure.md

```diff
@@ -0,0 +1,271 @@
+# ADR 001: Documentation Structure
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR proposes a documentation strategy based on the *Grand Unified Theory of Documentation* (David Laing) as described by [Divio](https://documentation.divio.com/).
+
+It outlines four specific use cases for documentation. Based on these use cases and other non-functional requirements, a structure is proposed that will address these concerns using GitHub as the Content Management System. 
+
+In addition it also proposes 
+
+- the use - and re-use - of document and format templates
+- specific codeowners for documentation 
+- comment and commit templates combined with githook checks
+
+The outcome shall be focused, consistent, high quality documentation. 
+
+## Context
+
+Good documentation is important to the success of software projects.
+
+*Writing excellent code doesn't end when your code compiles or even if your test coverage reaches 100%. It's easy to write something a computer understands, it's much harder to write something both a human and a computer understand. Your mission as a Code Health-conscious engineer is to write for humans first, computers second. Documentation is an important part of this skill.* [Google Documentation Best Practice](https://google.github.io/styleguide/docguide/best_practices.html)
+
+The documentation use cases, as outlined by Divio are:
+
+- Allow a new user to get started
+- Show a user how to solve a specific problem
+- Describe the machinery i.e. classes, functions, interfaces, parameters etc.
+- Explanation/context for design, scope etc
+
+![Documentation Quadrants](https://documentation.divio.com/_images/overview.png) 
+
+It is key that the documentation is structured and written such that:
+
+- Depending on use case the documentation can be found easily
+- The documenation is written in an appropriate style for the use case
+- Each type of documentation is written consistent style
+- Documentation is scoped to a specific use cases i.e. a tutorial doesn't describe (beyond a relevant link) why the software works, it just teaches
+
+Additional Documentation non-functional use cases include:
+
+- It SHOULD BE as close to the code as reasonably practicable
+- It SHOULD BE generated from code as much as possible
+- It SHOULD USE a consistent format 
+- It SHOULD BE useable from within the repository
+- It COULD HAVE an automatic process that converts teh content to a website based on [Read The Docs](https://readthedocs.com/), [Gitbook](https://www.gitbook.com/) or other suitable hosting system.
```

Comment:
```suggestion
- It COULD HAVE an automatic process that converts the content to a website based on [Read The Docs](https://readthedocs.com/), [Gitbook](https://www.gitbook.com/) or other suitable hosting systems.
```